### PR TITLE
- switches to a submodule for apidoctor

### PR DESCRIPTION
**Pull request by baywet**

### CODE REVIEW

**Comment by zengin** [comment]
Regurlar PR gates use only production packages, so I am kicking off end-to-end pipeline runs:

https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=60467&view=results
https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=60468&view=results

**Review by zengin — CHANGES_REQUESTED** [review]
there seems to be an error in project path. The end to end pipelines are failing.

**Comment by georgend** [comment]
I may have missed something, but why are we moving to using apidoctor as a submodule?

**Comment by baywet** [comment]
@georgend see this recap https://github.com/microsoftgraph/microsoft-graph-devx-api/pull/809#issue-1069687447

**Comment by georgend** [comment]
Good recap.

File: azure-pipelines/e2e-templates/checkouts.yml [hunk 1]
```diff
@@ -10,12 +10,6 @@ steps:
...
       displayName: checkout GE api
       fetchDepth: 1
-      persistCredentials: true
-
-  - ${{ if eq(parameters.shouldRunSnippetGeneration, true) }}:
-    - checkout: apidoctor
-      displayName: checkout apidoctor
-      fetchDepth: 1
       submodules: recursive
       persistCredentials: true
```
Comment (zengin) [inline azure-pipelines/e2e-templates/checkouts.yml:15]: argh! GitHub doesn't allow me to comment on lines that are not changed.

I think you need a `submodules: recursive` on the self checkout at the end of this file.

**Review by zengin — CHANGES_REQUESTED** [review]

File: azure-pipelines/e2e-templates/checkouts.yml (hunk 1 above)
Comment (baywet) [reply azure-pipelines/e2e-templates/checkouts.yml:15]: right, I got confused as I was working on devx api at the same time, thanks for pointing this out.

**Review by zengin — APPROVED** [review]

**Pull request merged**